tests/e2e/helpers/common.ts

```diff
@@ -66,9 +66,14 @@ export async function setupPlayground(
 ) {
   const monetizationCallback = spy<[window.MonetizationEvent], void>();
   await page.exposeFunction('monetizationCallback', monetizationCallback);
-  await page.goto(playgroundUrl(...walletAddressUrls));
-  await page.evaluate(() => {
-    window.addEventListener('monetization', monetizationCallback);
+  await page.addInitScript({
+    content: `window.addEventListener('monetization', monetizationCallback)`,
```

Comment:
`addInitScript` will be survive across `page.reload` and navigation (`exposeFunction` also survives such navigations), so we don't need to add listener again.